pkg/selector/physicalmachine/selector.go

```diff
@@ -212,7 +210,7 @@ func selectSpecifiedPhysicalMachines(ctx context.Context, c client.Client, spec
 	for ns, names := range spec.PhysicalMachines {
 		if !clusterScoped {
 			if targetNamespace != ns {
-				log.Info("skip namespace because ns is out of scope within namespace scoped mode", "namespace", ns)
+				log.L().WithName("physical-machine-selector").Info("skip namespace because ns is out of scope within namespace scoped mode", "namespace", ns)
```

Comment:
Hi, as the issue #2887 said, please move package level variable `log` as a field of `SelectImpl` struct, but not use the global logger.

Reply:
+1 PTAL @basit9958

Reply:
I'm on it